apps/veil/src/pages/tournament/server/previous-epochs.ts

```diff
@@ -0,0 +1,147 @@
+import { sql } from 'kysely';
+import { NextRequest, NextResponse } from 'next/server';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { AssetId, Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { serialize, Serialized } from '@/shared/utils/serializer';
+import { Gauge } from '@/shared/database/schema';
+import { pindexerDb } from '@/shared/database/client';
+import { base64ToUint8Array } from '@penumbra-zone/types/base64';
+
+const SORT_KEYS = ['epoch', 'lpReward', 'votingReward', ''] as const;
+export type PreviousEpochsSortKey = (typeof SORT_KEYS)[number];
+
+const DIRECTIONS = ['asc', 'desc'] as const;
+export type PreviousEpochsSortDirection = (typeof DIRECTIONS)[number];
+
+export interface PreviousEpochsRequest {
+  limit: number;
+  page: number;
+  sortKey: PreviousEpochsSortKey;
+  sortDirection: PreviousEpochsSortDirection;
+}
+
+export interface MappedGauge extends Omit<Gauge, 'asset_id'> {
+  asset: Metadata;
+}
+
+export interface PreviousEpochData {
+  epoch: number;
+  gauge: MappedGauge[];
+}
+
+export interface PreviousEpochsApiResponse {
+  total: number;
+  data: PreviousEpochData[];
+}
+
+const DEFAULT_LIMIT = 10;
+
+export const getQueryParams = (req: NextRequest): PreviousEpochsRequest => {
+  const { searchParams } = new URL(req.url);
+
+  const limit = Number(searchParams.get('limit')) || DEFAULT_LIMIT;
+  const page = Number(searchParams.get('page')) || 1;
+
+  const sortKeyParam = searchParams.get('sortKey');
+  const sortKey =
+    sortKeyParam && SORT_KEYS.includes(sortKeyParam as PreviousEpochsSortKey)
+      ? (sortKeyParam as PreviousEpochsSortKey)
+      : 'epoch';
+
+  const sortDirectionParam = searchParams.get('sortDirection');
+  const sortDirection =
+    sortDirectionParam && DIRECTIONS.includes(sortDirectionParam as PreviousEpochsSortDirection)
+      ? (sortDirectionParam as PreviousEpochsSortDirection)
+      : 'desc';
+
+  return {
+    limit,
+    page,
+    sortKey,
+    sortDirection,
+  };
+};
+
+const previousEpochsQuery = async ({ limit, page, sortDirection }: PreviousEpochsRequest) => {
+  // 1. Take the 'gauge' table and sort it by 'epoch' and then by 'portion', map asset_id to base64
+  const sortedGauge = pindexerDb
```

Comment:
note: using `pindexerDb` variable instead of `pindexer` class to move away from this class as it become too big and needs splitting into smaller files. Also the DB usage is now placed right where it needs to be used.

Reply:
Yeah, I think the pindexer class is superfluous, the individual api routes can be responsible for making their own queries.